src/bda/plone/discount/calculator.py

```diff
@@ -90,18 +96,30 @@ class GroupCartRulesLookup(RuleLookup):
     settings_iface = IGroupCartDiscountSettings
     for_attribute = FOR_GROUP
 
+class CouponCartRulesLookup(RuleLookup):
+    settings_iface = ICouponCartDiscountSettings
+    for_attribute = FOR_COUPON
 
 class RuleAcquierer(object):
     lookup_factory = None
     user_lookup_factory = None
     group_lookup_factory = None
+    coupon_lookup_factory = None
 
     def __init__(self, context):
         self.context = context
         self.date = datetime.now()
         self.member = api.user.get_current()
         self.user = None
         self.groups = None
+        
+        request = getRequest()
```

Comment:
Maybe extract this section in a separate function? I we wanted to be super flexible, you could register a utility which would get the discount code (if someone wanted to get the coupon code from somewhere else).